src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -406,7 +412,67 @@ internal T WithProofOfPosession(HttpRequestMessage httpRequestMessage, IPoPCrypt
             CommonParameters.AddApiTelemetryFeature(ApiTelemetryFeature.WithPoPScheme);
             CommonParameters.AuthenticationScheme = new PoPAuthenticationScheme(httpRequestMessage, popCryptoProvider);
 
-            return this as T;
+            return (T)this;
+        }
+
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item> An Authentication header is automatically added to the request</item>
+        /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
+        /// <item>MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</item>
+        /// </list>
+        /// </remarks>
+        public T WithProofOfPosession()
+        {
+            CommonParameters.UsingProofOfPossesion = true;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// Specifies the HTTP method of the HTTP request to the protected resource which requires a PoP token. 
+        /// The PoP token will be cryptographically bound to the request.
+        /// </summary>
+        /// <param name="httpMethod">Http Method for proof of possesion request.</param>
+        /// <returns></returns>
+        public T WithProofOfPosessionMethod(HttpMethod httpMethod)
+        {
+            CommonParameters.PopMethod = httpMethod;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// Specifies the URL of the HTTP request to the protected resource which requires a PoP token. 
+        /// The PoP token will be cryptographically bound to the request.
+        /// </summary>
+        /// <param name="uri">Protected resource URL.</param>
+        /// <returns></returns>
+        public T WithProofOfPosessionUri(Uri uri)
+        {
+            CommonParameters.PopUri = uri;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// Specifies a provider that can handle the asymmetric key operations needed by POP, that encapsulates a pair of 
+        /// public and private keys and some typical crypto operations.
+        /// All symetric operations are SHA256
+        /// </summary>
+        /// <param name="popCryptoProvider"> Proof of posession cryptography provider</param>
+        /// <returns></returns>
+        public T WithPopCryptoProvider(IPoPCryptoProvider popCryptoProvider)
```

Comment:
I think you are trying to tackle 2 stories at the same time, which makes it difficult to implement and review: 

1. Expose the crypto details to the user
2. Make the method and uri optional

I don't think you should tackle #2 right now, nobody asked for it and it will complicate things.